src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonElement.Parse.cs

```diff
@@ -73,14 +73,47 @@ internal static JsonElement ParseValue(Stream utf8Json, JsonDocumentOptions opti
             return document.RootElement;
         }
 
-        internal static JsonElement ParseValue(ReadOnlySpan<byte> utf8Json, JsonDocumentOptions options)
+        /// <summary>
+        /// Parses UTF8-encoded text representing a single JSON value into a <see cref="JsonElement"/>.
+        /// </summary>
+        /// <param name="utf8Json">The JSON text to parse.</param>
+        /// <param name="options">Options to control the reader behavior during parsing.</param>
+        /// <returns>A <see cref="JsonElement"/> representation of the JSON value.</returns>
+        /// <exception cref="JsonException"><paramref name="utf8Json"/> does not represent a valid single JSON value.</exception>
+        /// <exception cref="ArgumentException"><paramref name="options"/> contains unsupported options.</exception>
+        public static JsonElement Parse([StringSyntax(StringSyntaxAttribute.Json)] ReadOnlySpan<byte> utf8Json, JsonDocumentOptions options = default)
         {
             JsonDocument document = JsonDocument.ParseValue(utf8Json, options);
             return document.RootElement;
         }
 
-        internal static JsonElement ParseValue(string json, JsonDocumentOptions options)
+        /// <summary>
+        /// Parses text representing a single JSON value into a <see cref="JsonElement"/>.
+        /// </summary>
+        /// <param name="json">The JSON text to parse.</param>
+        /// <param name="options">Options to control the reader behavior during parsing.</param>
+        /// <returns>A <see cref="JsonElement"/> representation of the JSON value.</returns>
+        /// <exception cref="JsonException"><paramref name="json"/> does not represent a valid single JSON value.</exception>
+        /// <exception cref="ArgumentException"><paramref name="options"/> contains unsupported options.</exception>
+        public static JsonElement Parse([StringSyntax(StringSyntaxAttribute.Json)] ReadOnlySpan<char> json, JsonDocumentOptions options = default)
+        {
+            JsonDocument document = JsonDocument.ParseValue(json, options);
```

Comment:
You're creating a disposable object without a `using`.  Isn't this a memory leak? I suppose it'll get cleaned up by the GC, but wouldn't that mean the memory that backs the element could be reclaimed at any time?  What guarantees do we have about this element's persistence?

Reply:
The `ParseValue` methods don't hold pooled resources:

https://github.com/dotnet/runtime/blob/9214279d93b8b422495a98eb4edda91e92bd60c3/src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs#L320-L321

Reply:
I don't understand the relevance of that, but @stephentoub's answer in the main thread resolves my concern anyhow.

Reply:
> I don't understand the relevance of that, but @stephentoub's answer in the main thread resolves my concern anyhow.

JsonDocument is disposable because the byte[] it holds onto _may_ have come from ArrayPool, in which case we'd like to return the array to the pool when it's no longer needed for the document. Not doing so isn't a leak, it just means we lose out on the potential efficiency gains of being able to reuse that array when someone else asks the pool for one (if one isn't available, the pool will need to allocate a new one). The method Pranav highlights is constructing a document with a new array rather than one from the pool.